File: SDKs/googleapis.com/run/v1/python/src/sdk/models/shared/revisionspec.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import container as shared_container
from ..shared import localobjectreference as shared_localobjectreference
from ..shared import volume as shared_volume


@dataclass_json
@dataclasses.dataclass
class RevisionSpec:
    r"""RevisionSpec
    RevisionSpec holds the desired state of the Revision (from the client).
    """
    
    container_concurrency: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('containerConcurrency') }})
    containers: Optional[list[shared_container.Container]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('containers') }})
    enable_service_links: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('enableServiceLinks') }})
    image_pull_secrets: Optional[list[shared_localobjectreference.LocalObjectReference]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('imagePullSecrets') }})
    service_account_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('serviceAccountName') }})
    timeout_seconds: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('timeoutSeconds') }})
    volumes: Optional[list[shared_volume.Volume]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('volumes') }})
    

File: SDKs/googleapis.com/run/v1/java/lib/src/main/java/openapisdk/models/shared/RevisionSpec.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * RevisionSpec
 * RevisionSpec holds the desired state of the Revision (from the client).
**/
public class RevisionSpec {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("containerConcurrency")
    public Integer containerConcurrency;
    public RevisionSpec withContainerConcurrency(Integer containerConcurrency) {
        this.containerConcurrency = containerConcurrency;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("containers")
    public Container[] containers;
    public RevisionSpec withContainers(Container[] containers) {
        this.containers = containers;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("enableServiceLinks")
    public Boolean enableServiceLinks;
    public RevisionSpec withEnableServiceLinks(Boolean enableServiceLinks) {
        this.enableServiceLinks = enableServiceLinks;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("imagePullSecrets")
    public LocalObjectReference[] imagePullSecrets;
    public RevisionSpec withImagePullSecrets(LocalObjectReference[] imagePullSecrets) {
        this.imagePullSecrets = imagePullSecrets;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("serviceAccountName")
    public String serviceAccountName;
    public RevisionSpec withServiceAccountName(String serviceAccountName) {
        this.serviceAccountName = serviceAccountName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("timeoutSeconds")
    public Integer timeoutSeconds;
    public RevisionSpec withTimeoutSeconds(Integer timeoutSeconds) {
        this.timeoutSeconds = timeoutSeconds;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("volumes")
    public Volume[] volumes;
    public RevisionSpec withVolumes(Volume[] volumes) {
        this.volumes = volumes;
        return this;
    }
}